### PR TITLE
[statsdreceiver] add obsreport metrics for accepted/refused count

### PR DESCRIPTION
**Pull request by hardproblems**
**Description:** <Describe what has changed.>
Add accepted/refused metrics to statsdreceiver

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24278

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

cc @jmacd @dmitryax

### CODE REVIEW

File: receiver/statsdreceiver/receiver.go [hunk 1]
```diff
@@ -110,10 +122,13 @@ func (r *statsdReceiver) Start(ctx context.Context, _ component.Host) error {
 				batchMetrics := r.parser.GetMetrics()
 				for _, batch := range batchMetrics {
 					batchCtx := client.NewContext(ctx, batch.Info)
-
-					if err := r.Flush(batchCtx, batch.Metrics, r.nextConsumer); err != nil {
+					numPoints := batch.Metrics.DataPointCount()
+					r.obsrecv.StartMetricsOp(ctx)
```
Comment (TylerHelmuth) [inline receiver/statsdreceiver/receiver.go:126]: This function should take the `batchCtx` as the input context and the returned context should be used in `Flush`, and `EndMetricsOp`

Comment (hardproblems) [reply receiver/statsdreceiver/receiver.go]: fixed PTAL

**Comment by sirianni** [comment]
@TylerHelmuth FYI see also a related PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31839

File: receiver/statsdreceiver/receiver.go [hunk 2]
```diff
@@ -110,10 +122,13 @@ func (r *statsdReceiver) Start(ctx context.Context, _ component.Host) error {
 				batchMetrics := r.parser.GetMetrics()
 				for _, batch := range batchMetrics {
 					batchCtx := client.NewContext(ctx, batch.Info)
-
-					if err := r.Flush(batchCtx, batch.Metrics, r.nextConsumer); err != nil {
+					numPoints := batch.Metrics.DataPointCount()
+					r.obsrecv.StartMetricsOp(batchCtx)
```
Comment (TylerHelmuth) [inline receiver/statsdreceiver/receiver.go:126]: This function returns a `context.Context` and thats the context that should be used in `Flush` and `EndMetricsOp`

Comment (hardproblems) [reply receiver/statsdreceiver/receiver.go]: oops didn't understand the comment previously. updated again. PTAL

**Review by TylerHelmuth — COMMENTED** [review]
@hardproblems have you run this locally and confirmed the new metrics appear as expected?

**Comment by hardproblems** [comment]
> @hardproblems have you run this locally and confirmed the new metrics appear as expected?
@TylerHelmuth actually no metrics appeared. I'm going to try ctx.Background()

**Comment by TylerHelmuth** [comment]
Make sure you send some data through, I don't believe `0` will get reported

**Pull request closed without merging**